main.js

```diff
@@ -7,120 +7,147 @@ const { addExtra } = require("puppeteer-extra");
-		  });
+    Puppeteer.use(
+        Recaptcha({
+            provider: { id: "2captcha", token: process.env.RECAPTCHATOKEN },
+        })
+    );
 
-	const gatherData = async () => {
-		const results = await Promise.all(
-			scrapers.map((scraper) =>
-				scraper(browser).catch((error) => {
-					//print out the issue but don't fail, this way we still publish updates
-					//for other locations even if this website's scrape doesn't work
-					console.log(error);
-					return null;
-				})
-			)
-		);
-		browser.close();
-		const finalResultsArray = [];
-		for (const result of results) {
-			if (Array.isArray(result)) {
-				finalResultsArray.push(...result);
-			} else if (result) {
-				//ignore nulls
-				finalResultsArray.push(result);
-			}
-		}
+    const browser = process.env.DEVELOPMENT
+        ? await Puppeteer.launch({
+              executablePath: process.env.CHROMEPATH,
+          })
+        : await Puppeteer.launch({
+              args: chromium.args,
+              defaultViewport: chromium.defaultViewport,
+              executablePath: await chromium.executablePath,
+              headless: chromium.headless,
+              ignoreHTTPSErrors: true,
+          });
 
-		const responseJson = {
-			results: finalResultsArray,
-		};
+    const gatherData = async () => {
+        const results = await Promise.all(
+            scrapers.map((scraper) =>
+                scraper(browser).catch((error) => {
+                    //print out the issue but don't fail, this way we still publish updates
+                    //for other locations even if this website's scrape doesn't work
+                    console.log(error);
+                    return null;
+                })
+            )
+        );
+        browser.close();
+        let scrapedResultsArray = [];
+        for (const result of results) {
+            if (Array.isArray(result)) {
+                scrapedResultsArray.push(...result);
+            } else if (result) {
+                //ignore nulls
+                scrapedResultsArray.push(result);
+            }
+        }
 
-		if (process.env.DEVELOPMENT) {
-			console.log("The following data would be published:");
-			console.dir(responseJson, { depth: null });
-			fs = require("fs");
-			await new Promise((resolve, reject) => {
-				fs.writeFile("out.json", JSON.stringify(responseJson), (err) => {
-					if (err) {
-						reject(err);
-					} else {
-						resolve();
-					}
-				});
-			});
-			return;
-		} else {
-			const params = {
-				Bucket: process.env.AWSS3BUCKETNAME,
-				Key: "data.json",
-				Body: JSON.stringify(responseJson),
-			};
+        let finalResultsArray = [];
+        if (process.argv.length <= 2) {
+            //only add default data if we're not testing individual scrapers
+            finalResultsArray = dataDefaulter.addDefaultsToResults(
+                scrapedResultsArray,
+                cachedResults,
+                15 * 60 //15 minutes allowance for cached data
```

Comment:
Can we move this to the top as a constant?  I'm not sure what the right JS pattern is here.  Or make configurable?